### PR TITLE
Allow a conn open ack to succeed in the happy case

### PR DESCRIPTION
**Pull request by vitorenesduarte**
Closes: #699 

## Description

Currently, the conn open ack handler requires a connection to have the counterparty connection id set (line 36):

https://github.com/informalsystems/ibc-rs/blob/02aef9cf5055a3817579b555c84289b99d5d08a8/modules/src/ics03_connection/handler/conn_open_ack.rs#L33-L45

However, if the connection is in the `Init` state (the happy/normal case), the counterparty connection id is unknown (i.e. a `None`), in which case the conn open ack fails with a `ConnectionMismatch`.

With this PR, we allow the counterparty connection id to be `None`, and only fail with a `ConnectionMismatch` in case it's a `Some` (that doesn't match).

Since the`MsgConnectionOpenAck.counterparty_connection_id` field is always required, this PR also changes it to simply be a `ConnectionId` (instead of an `Option<ConnectionId>`).

______

For contributor use:

- [ ] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: modules/src/ics03_connection/handler/conn_open_ack.rs [hunk 1]
```diff
@@ -168,16 +177,6 @@ mod tests {
             CommitmentPrefix::from(vec![]), // incorrect field
         ));
 
-        // A connection end with correct state & prefix, but incorrect counterparty; exercises
-        // unsuccessful processing path.
-        let mut conn_end_cparty = conn_end_open.clone();
-        conn_end_cparty.set_state(State::Init);
-        conn_end_cparty.set_counterparty(Counterparty::new(
-            client_id.clone(),
-            None, // incorrect field
```
Comment (vitorenesduarte) [inline modules/src/ics03_connection/handler/conn_open_ack.rs:177]: This test has been removed as my understanding is that `msg.counterparty_connection_id` should always be a `Some` (which hints that it should not be an `Option`).

File: modules/src/ics03_connection/handler/verify.rs [hunk 2]
```diff
@@ -96,7 +96,7 @@ pub fn verify_connection_proof(
             proof_height,
             connection_end.counterparty().prefix(),
             proof,
-            &connection_end.counterparty().connection_id().unwrap(),
+            connection_end.counterparty().connection_id(),
```
Comment (vitorenesduarte) [inline modules/src/ics03_connection/handler/verify.rs:99]: If the connection is in the `Init` state, `connection_end.counterparty().connection_id()` is a `None`, and thus it's not okay to `unwrap`.

File: modules/src/ics03_connection/handler/conn_open_ack.rs (hunk 1 above)
Comment (vitorenesduarte) [reply modules/src/ics03_connection/handler/conn_open_ack.rs:177]: Now this field is no longer an `Option`.

**Review by adizere — APPROVED** [review]
LGTM, thank you Vitor.

**Pull request merged**